File: JScript/clsPfSupport.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//



//Define Class
function clsPfSupport()
{
  key =0;
  js  = 0;
  rx = 0;          //.. joint X directional restraint ..
  ry = 0;          //.. joint Y directional restraint ..
  rm = 0;          //.. joint Z rotational restraint ..
}


clsPfSupport.prototype.initialise = function()
{
  this.key = 0;
  this.js = 0;
  this.rx = 0;
  this.ry = 0;
  this.rm = 0;
}

clsPfSupport.prototype.setValues = function(supportKey, SupportNode, RestraintX, RestraintY, RestraintMoment)
{
  this.key = supportKey
  this.js = SupportNode
  this.rx = RestraintX
  this.ry = RestraintY
  this.rm = RestraintMoment
}


clsPfSupport.prototype.sprint = function()
{
  var s;
  
  s = "";
  s = s + StrLPad(this.key.toString(), 8);
  s = s + StrLPad(this.js.toString(), 6);
  s = s + StrLPad(this.rx.toString(), 6);
  s = s + StrLPad(this.ry.toString(), 6);
  s = s + StrLPad(this.rm.toString(), 6);
           
  return s         
}

clsPfSupport.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
}

clsPfSupport.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}


clsPfSupport.prototype.fgetData = function(fp)
{
  var s;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "fgetData ...");
  
  s = fp.ReadLine();
  WScript.Echo( s);
  this.sgetData(s);
   
  WScript.Echo( "... fgetData");
}


clsPfSupport.prototype.sgetData = function(s)
{
  var n;
  var dataflds;
  var i,n;

   WScript.Echo( "sgetData ...");
   WScript.Echo( s);

   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
   dataflds = s.match(/-?\d+(?:[,.]\d+)?/gi);
   
   
   n = dataflds.length;
   for (i=0;i<n;i++)
   {
    WScript.Echo( i + "<" + dataflds[i] + ">" );
   }
   
    this.key = parseInt(dataflds[0]);
    this.js = parseFloat(dataflds[1]);
    this.rx = parseFloat(dataflds[2]);
    this.ry = parseFloat(dataflds[3]);
    this.rm = parseFloat(dataflds[4]);
    
   
   
   WScript.Echo( "... sgetData");
}

File: SharpDevelop/CSharp/w7cpframe/clsPfSupport.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;

namespace w7cpframe
{
    class clsPfSupport
    {


        public int key = 0;
        public int js = 0;
        public int rx = 0;          //.. joint X directional restraint ..
        public int ry = 0;          //.. joint Y directional restraint ..
        public int rm = 0;          //.. joint Z rotational restraint ..



        public void initialise()
        {
            key = 0;
            js = 0;
            rx = 0;
            ry = 0;
            rm = 0;
        }

        public void setValues(int supportKey, int SupportNode, int RestraintX, int RestraintY, int RestraintMoment)
        {
            key = supportKey;
            js = SupportNode;
            rx = RestraintX;
            ry = RestraintY;
            rm = RestraintMoment;
        }


        public string sprint()
        {
            string s;

            s = "";
            s = s + key.ToString().PadLeft(8);
            s = s + js.ToString().PadLeft(6);
            s = s + rx.ToString().PadLeft(6);
            s = s + ry.ToString().PadLeft(6);
            s = s + rm.ToString().PadLeft(6);

            return s;
        }

        public void cprint()
        {
            System.Console.WriteLine(sprint());
        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }


        public void fgetData(StreamReader fp)
        {
            string s;

            System.Console.WriteLine("fgetData ...");

            s = fp.ReadLine();
            System.Console.WriteLine(s);
            sgetData(s);

            System.Console.WriteLine("... fgetData");
        }


        public void sgetData(string s)
        {
            string [] dataflds = new string[10]; //(0 To 9);
            char [] delimters = new char[] { ' ', '\t' };
            int i, n;

            System.Console.WriteLine("sgetData ...");
            System.Console.WriteLine(s);

            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s, pattern, "");

            dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine(match.Value);
                    i++;
                }
            }


            n = dataflds.Length;
            for (i = 0; i < n; i++)
            {
                System.Console.WriteLine(i + "<" + dataflds[i] + ">");
            }

            key = int.Parse(dataflds[0]);
            js = int.Parse(dataflds[1]);
            rx = int.Parse(dataflds[2]);
            ry = int.Parse(dataflds[3]);
            rm = int.Parse(dataflds[4]);



            System.Console.WriteLine("... sgetData");
        }




    } //class
} //name space
